scr/main/java/bittercode/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import javax.servlet.http.HttpSession;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.UsersDBConstants;
+import com.bittercode.model.StoreException;
+import com.bittercode.model.User;
+import com.bittercode.model.UserRole;
+import com.bittercode.service.UserService;
+import com.bittercode.util.DBUtil;
+
+public class UserServiceImpl implements UserService {
+
+    private static final String registerUserQuery = "INSERT INTO " + UsersDBConstants.TABLE_USERS
+            + "  VALUES(?,?,?,?,?,?,?,?)";
+
+    private static final String loginUserQuery = "SELECT * FROM " + UsersDBConstants.TABLE_USERS + " WHERE "
+            + UsersDBConstants.COLUMN_USERNAME + "=? AND " + UsersDBConstants.COLUMN_PASSWORD + "=? AND "
+            + UsersDBConstants.COLUMN_USERTYPE + "=?";
+
+    @Override
+    public User login(UserRole role, String email, String password, HttpSession session) throws StoreException {
+        Connection con = DBUtil.getConnection();
+        PreparedStatement ps;
+        User user = null;
+        try {
+            String userType = UserRole.SELLER.equals(role) ? "1" : "2";
+            ps = con.prepareStatement(loginUserQuery);
+            ps.setString(1, email);
+            ps.setString(2, password);
+            ps.setString(3, userType);
+            ResultSet rs = ps.executeQuery();
+            if (rs.next()) {
+                user = new User();
+                user.setFirstName(rs.getString("firstName"));
+                user.setLastName(rs.getString("lastName"));
+                user.setPhone(rs.getLong("phone"));
+                user.setEmailId(email);
+                user.setPassword(password);
+                session.setAttribute(role.toString(), user.getEmailId());
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider improved SQL exception handling</b></div>
<div id="fix">

Consider handling `SQLException` more gracefully by throwing a custom `StoreException` with appropriate error details instead of just printing the stack trace. This would allow better error handling at higher layers.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -48,6 +48,6 @@
          } catch (SQLException e) {
 -            e.printStackTrace();
 +            throw new StoreException(ResponseCode.DATABASE_CONNECTION_FAILURE, e);
          }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2517477681>#c23dbc</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged